### PR TITLE
Fix for "Resend confirmation email" button to use the correct email address

### PR DESCRIPTION
**Pull request by agr**
Attempt to address https://github.com/NuGet/Engineering/issues/2512

### CODE REVIEW

File: src/NuGetGallery/Controllers/AccountsController.cs [hunk 1]
```diff
@@ -116,7 +116,14 @@ public virtual async Task<ActionResult> ConfirmationRequiredPost(string accountN
             ConfirmationViewModel model;
             if (!alreadyConfirmed)
```
Comment (scottbommarito) [inline src/NuGetGallery/Controllers/AccountsController.cs]: IMO this variable name is confusing

I would either
- inline the check (`account.UnconfirmedEmailAddress != null`)
- rename the variable to `hasUnconfirmedEmailAddress`

File: tests/NuGetGallery.Facts/Controllers/OrganizationsControllerFacts.cs [hunk 2]
```diff
@@ -273,6 +273,24 @@ public async Task WithNonOrganizationAdmin_ReturnsForbidden(Func<Fakes, User> ge
                 Assert.NotNull(result);
                 Assert.Equal((int)HttpStatusCode.Forbidden, result.StatusCode);
             }
+
+            [Fact]
+            public async Task ResendsProperConfirmationEmail()
```
Comment (scottbommarito) [inline tests/NuGetGallery.Facts/Controllers/OrganizationsControllerFacts.cs]: We should add a test as well that verifies that a `NewAccountMessage` is sent when `account.EmailAddress` is null.

Comment (agr) [reply tests/NuGetGallery.Facts/Controllers/OrganizationsControllerFacts.cs]: Done

File: src/NuGetGallery/Controllers/AccountsController.cs (hunk 1 above)
Comment (agr) [reply src/NuGetGallery/Controllers/AccountsController.cs]: Renamed the variable.

File: src/NuGetGallery/Controllers/AccountsController.cs [hunk 3]
```diff
@@ -111,12 +111,19 @@ public virtual async Task<ActionResult> ConfirmationRequiredPost(string accountN
                 return new HttpStatusCodeResult(HttpStatusCode.Forbidden, Strings.Unauthorized);
             }
 
-            var alreadyConfirmed = account.UnconfirmedEmailAddress == null;
+            var hasUnconfirmedEmailAddress = account.UnconfirmedEmailAddress == null;
```
Comment (cristinamanum) [inline src/NuGetGallery/Controllers/AccountsController.cs]: When the account.UnconfirmedEmailAddress == null it means that the email is confirmed. Why the name changed?

Comment (agr) [reply src/NuGetGallery/Controllers/AccountsController.cs]: The rename was suggested by @scottbommarito: https://github.com/NuGet/NuGetGallery/pull/7273#discussion_r295047300

**Pull request merged**